src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -2955,12 +2965,12 @@ public boolean tokadd_ident(int c) {
      * Because this version does not use a separate token buffer we only just increment lex_p.  When we reach
      * end of the token it will just get the bytes directly from source directly.
      */
-    public boolean tokadd_mbchar(int first_byte) {
+    public boolean tokadd_mbchar(int firstByte) {
         int length = precise_mbclen();
 
-        if (length <= 0) {
+        if (length <= 0 || lexb.getEncoding() == USASCIIEncoding.INSTANCE && firstByte > 127) {
```

Comment:
Another place to use the `!isASCII` maybe?

Reply:
Why does this error trigger only if the encoding is exactly US-ASCII?
Should it also handle other encodings if length <= 0?

Reply:
This is bad code. I thought I deleted it and missed it in the diff. The case is now handled in `precise_mbclen` checking if `lexb` has a `CR_BROKEN` code range.